src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -55,19 +57,22 @@ internal abstract class AbstractMakeMethodAsynchronousCodeFixProvider : CodeFixP
             {
                 context.RegisterCodeFix(
                     new MyCodeAction(GetMakeAsyncTaskFunctionResource(), c => FixNodeAsync(
-                        context.Document, diagnostic, keepVoid: false, cancellationToken: c)),
+                        context.Document, diagnostic, keepVoid: false, isEntryPoint, cancellationToken: c)),
                     context.Diagnostics);
 
-                context.RegisterCodeFix(
-                    new MyCodeAction(GetMakeAsyncVoidFunctionResource(), c => FixNodeAsync(
-                        context.Document, diagnostic, keepVoid: true, cancellationToken: c)),
-                    context.Diagnostics);
+                if (!isEntryPoint)
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(GetMakeAsyncVoidFunctionResource(), c => FixNodeAsync(
+                            context.Document, diagnostic, keepVoid: true, isEntryPoint: false, cancellationToken: c)),
+                        context.Diagnostics);
+                }
             }
             else
             {
                 context.RegisterCodeFix(
                     new MyCodeAction(GetMakeAsyncTaskFunctionResource(), c => FixNodeAsync(
-                        context.Document, diagnostic, keepVoid: false, cancellationToken: c)),
+                        context.Document, diagnostic, keepVoid: false, isEntryPoint, cancellationToken: c)),
                     context.Diagnostics);
```

Comment:
nit: it looks to me like this code is exactly the same as in the other branch. could this be simplified to offer this fix unconditionally and then have a check `if (isOrdinaryOrLocalFunction && symbol.ReturnsVoid && !isEntryPoint)` to offer the "keep void" fix?

Reply:
Thanks!